src/ja/common/message/worker_commands/start_job.py

```diff
@@ -26,12 +27,17 @@ def job(self) -> Job:
         """
         return self._job
 
-    def execute(self, worker_client: JobWorker) -> WorkerResponse:
+    def execute(self, docker_interface: DockerInterface) -> Response:
         """!
         Start the job on the worker machine using the provided @worker_client
-        @param worker_client:  the Worker object to use for the execution
+        @param docker_interface: the docker interface to use for the execution.
         @return: a WorkerResponse with the appropriate response
         """
+        try:
+            docker_interface.add_job(job=self.job)
+            return Response(self.RESPONSE_SUCCESS % (self.job.uid, docker_interface.worker_uid), is_success=True)
+        except ValueError:
```

Comment:
Is this possible? Wouldn't it be rather impractical if a job cannot be added to a machine just because the uid happens to exist there already?

Reply:
This should not be a problem because

1. workers only track Jobs that are currently dispatched to them and
2. Job UIDs are unique anyways.

If a worker retrieves two Jobs with the same UIDs it's in all likelihood due to a bug.